tools/gen-postmortem-metadata.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+
+#
+# gen-postmortem-metadata.py output_file.cc
+#
+# Creates debugging symbols to help naviage Node's internals using post-mortem
+# debugging tools.
+#
+
+import sys
+
+
+class DebugSymbol(object):
+    type_ = 'int'
+    _PREFIX = 'nodedbg_'
+
+    def __init__(self, name, value, headers=[], type_=None):
+        self.name = name
+        self.value = value
+        self.headers = headers
+        self.type_ = type_ or DebugSymbol.type_
+
+    @classmethod
+    def get_headers(cls, debug_symbols):
+        seen = set()
+        seen_add = seen.add  # faster than seen.add on each iteration
+        headers = [debug_symbol.headers for debug_symbol in debug_symbols]
+        headers = sum(headers, [])
+
+        return [h for h in headers if not (h in seen or seen_add(h))]
```

Comment:
This seems like a convoluted way of doing:
```python
headers = set()
for d in debug_symbols:
  headers.update(d.headers)
return list(headers)
```
Or even just:
```python
return list(set(sum(d.headers for d in debug_symbols, [])))
```

Reply:
I prefer those ways too, but the order the headers were declared should be preserved (some headers need to be included before others). I'll add a comment on this method explaining that.